db/migrations/20161026_1148_add_secondary_index_to_incidents.rb

```diff
@@ -0,0 +1,50 @@
+# Migration to add a secondary index to the Incidents table.
+class AddSecondaryIndexToIncidents < DynamoDB::Migration::Unit
+  def update
+    logger = Logger.new(STDOUT)
+
+    logger.info "Running migration #{self.class.name}:"
+
+    client.update_table(
+      table_name: "#{Dynamoid.config.namespace}_incidents",
+      attribute_definitions: [
+        {
+          attribute_name: "id",
+          attribute_type: "S"
+        },
+        {
+          attribute_name: "ursus_id_str",
+          attribute_type: "S"
+        }
+      ],
+      provisioned_throughput: {
+        read_capacity_units: Dynamoid::Config.read_capacity,
+        write_capacity_units: Dynamoid::Config.write_capacity
+      },
+      global_secondary_index_updates: [
+        {
+          create: {
+            index_name: Incident.indexes['ursus_id_str'].name,
+            key_schema: [
+              {
+                attribute_name: "ursus_id_str",
+                key_type: "HASH"
+              }
+            ],
+            projection: {
```

Comment:
The projection specifies which attributes of the original table end up in the secondary key table. I could do `projection_type: "ALL"` to include all attributes, but that would be wasteful since we actually only need the "id" attribute for our purposes.